test/simplify_qdq_test.cpp

```diff
@@ -1148,4 +1154,64 @@ TEST_CASE(dot_correctness)
     EXPECT(migraphx::verify::verify_rms_range(rv1, rv2));
 }
 
+TEST_CASE(dot_reused)
+{
+    migraphx::shape sh{migraphx::shape::float_type, {256, 256}};
+
+    migraphx::module m1;
+    {
+        auto x     = m1.add_parameter("x", sh);
+        auto y     = m1.add_parameter("y", sh);
+        auto w1    = m1.add_parameter("w1", sh);
+        auto w2    = m1.add_parameter("w2", sh);
+        auto scale = m1.add_literal(0.5f);
+        auto zero  = m1.add_literal(std::int8_t{0});
+
+        auto q1   = add_quantize_op(m1, "quantizelinear", x, scale, zero);
+        auto d1   = add_quantize_op(m1, "dequantizelinear", q1, scale, zero);
+        auto q2   = add_quantize_op(m1, "quantizelinear", w1, scale, zero);
+        auto d2   = add_quantize_op(m1, "dequantizelinear", q2, scale, zero);
+        auto dot1 = m1.add_instruction(migraphx::make_op("dot"), d1, d2);
+        auto add1 = m1.add_instruction(migraphx::make_op("add"), dot1, y);
+
+        auto q3   = add_quantize_op(m1, "quantizelinear", add1, scale, zero);
+        auto d3   = add_quantize_op(m1, "dequantizelinear", q3, scale, zero);
+        auto q4   = add_quantize_op(m1, "quantizelinear", w2, scale, zero);
+        auto d4   = add_quantize_op(m1, "dequantizelinear", q4, scale, zero);
+        auto dot2 = m1.add_instruction(migraphx::make_op("dot"), d3, d4);
+        auto add2 = m1.add_instruction(migraphx::make_op("add"), dot2, add1);
+        m1.add_return({add2});
+    }
+
+    migraphx::module m2;
+    {
+        auto x     = m2.add_parameter("x", sh);
+        auto y     = m2.add_parameter("y", sh);
+        auto w1    = m2.add_parameter("w1", sh);
+        auto w2    = m2.add_parameter("w2", sh);
+        auto scale = m2.add_literal(0.5f);
+        auto zero  = m2.add_literal(std::int8_t{0});
+
+        auto q1 = add_quantize_op(m2, "quantizelinear", x, scale, zero);
+        auto q2 = add_quantize_op(m2, "quantizelinear", w1, scale, zero);
+
+        auto dot1       = m2.add_instruction(migraphx::make_op("quant_dot"), q1, q2);
+        auto out_scale1 = add_scale_mul(m2, scale, scale, 1, 1, sh.lens());
+        auto d1         = add_quantize_op(m2, "dequantizelinear", dot1, out_scale1);
+        auto add1       = m2.add_instruction(migraphx::make_op("add"), d1, y);
+
+        auto q3         = add_quantize_op(m2, "quantizelinear", add1, scale, zero);
+        auto q4         = add_quantize_op(m2, "quantizelinear", w2, scale, zero);
+        auto dot2       = m2.add_instruction(migraphx::make_op("quant_dot"), q3, q4);
+        auto out_scale2 = add_scale_mul(m2, scale, scale, 1, 1, sh.lens());
+        auto d2         = add_quantize_op(m2, "dequantizelinear", dot2, out_scale2);
+        auto d3   = add_quantize_op(m2, "dequantizelinear", q3, q3->inputs()[1], q3->inputs()[2]);
```

Comment:
I dont quite understand why this is desirable. Seems to me like this just degrades accuracy by using a "fake quantized" result. Why do we want to add `d3` rather than just adding `add1` (which is already float)?

Reply:
Because then the qlinear op cannot be fused with add. Ideally we want the output of the activations to be int8 which will go into the next convolution or another pointwise if thats the case.

Reply:
Ok I see, I'm not totally familiar with all the fusion passes but I think this makes sense